packages/api/src/external/hie/carequality-analytics.ts

```diff
@@ -178,16 +182,56 @@ export function getOutboundDocQuerySuccessFailureCount(response: OutboundDocumen
 
 export function getOutboundDocRetrievalSuccessFailureCount(
   response: OutboundDocumentRetrievalResp[]
-): { successCount: number; failureCount: number } {
+): {
+  totalCount: number;
+  successCount: number;
+  failureCount: number;
+  noDocumentFoundCount: number;
+  schemaErrorCount: number;
+  httpErrorCount: number;
+  registryErrorCount: number;
+} {
   let successCount = 0;
   let failureCount = 0;
+  let totalCount = 0;
+  let noDocumentFoundCount = 0;
+  let schemaErrorCount = 0;
+  let httpErrorCount = 0;
+  let registryErrorCount = 0;
   for (const result of response) {
-    if (isSuccessfulOutboundDocRetrievalResponse(result)) {
-      successCount++;
-    } else if (result.operationOutcome?.issue) {
-      failureCount++;
+    if (result.documentReference) {
+      totalCount += result.documentReference.length;
+      successCount += result.documentReference.length;
+    }
+    if (result.operationOutcome?.issue) {
+      totalCount += result.operationOutcome.issue.filter(
+        issue => issue.id != undefined && issue.id.length > 0
+      ).length;
+      failureCount += result.operationOutcome.issue.filter(
```

Comment:
can't we have one pass thru `result.operationOutcome.issue`, and catch all the different cases right away? Why are we filtering the same thing a buttload of times? 😄